crypto/rand_extra/urandom_test.cc

```diff
@@ -193,6 +174,34 @@ static const unsigned GETRANDOM_ERROR = 16;
 static const unsigned URANDOM_ERROR = 32;
 static const unsigned NEXT_FLAG = 64;
 
+// ReadString parses string at address |addr| in child process |pid|.
+static std::string ReadString(pid_t pid, unsigned long addr) {
+  std::string result;
+  size_t i = 0;
+  while (i < 4096) { // Don't accept paths longer than this.
+    long data = ptrace(PTRACE_PEEKDATA, pid, addr + i, NULL);
+    if (data == -1 && errno) {
```

Comment:
what's `errno`??

Reply:
It's defined by the std library. It carries information about which error was observed by the system call: https://man7.org/linux/man-pages/man3/errno.3.html